[Doc] New section regarding Intel projects workflow

cristianmihaipopa opened the pull request:
## PR Description

New page that describes the workflow used in building the HDL Intel projects & the files that are involved in the process. This page is just a prototype, one more project needs to be included (Stratix10), but only after the major changes are made, if there will be any. 


## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the code style guidelines
- [x] I have performed a self-review of changes
- [ ] I have compiled all hdl projects and libraries affected by this PR
- [ ] I have tested in hardware affected projects, at least on relevant boards
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe files, Copyright etc)
- [ ] I have not introduced new Warnings/Critical Warnings on compilation
- [ ] I have added new hdl testbenches or updated existing ones

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+the larger application can run
+
+
+Below there're useful links who explain in more detail the process and the
+necessary components:
+
+-  `Booting and Configuration <https://www.intel.com/content/dam/support/us/en/programmable/support-resources/bulk-container/pdfs/literature/hb/arria-v/av-5400a.pdf#:~:text=This%20topic%20describes%20the%20booting%20of>`__
+-  `SocBootFromFPGA <https://community.intel.com/t5/FPGA-Wiki/SocBootFromFPGA/ta-p/735773#:~:text=Introduction.%20The%20HPS%20(Hardened%20Processor%20System)>`__
+-  `Intel Bootloader <https://www.intel.com/content/www/us/en/support/programmable/support-resources/design-guidance/soc-bootloader.html#:~:text=Intel%C2%AE%20SoC%20FPGAs%20use%20a%20bootloader>`__
```
```suggestion
In the following links you can find more details on the process and the necessary components:

-  `Booting and Configuration <https://www.intel.com/content/dam/support/us/en/programmable/support-resources/bulk-container/pdfs/literature/hb/arria-v/av-5400a.pdf>`__
-  `SocBootFromFPGA <https://community.intel.com/t5/FPGA-Wiki/SocBootFromFPGA/ta-p/735773>`__
-  `Intel Bootloader <https://www.intel.com/content/www/us/en/support/programmable/support-resources/design-guidance/soc-bootloader.html>`__
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+Necessary files for booting up a HDL project
+-------------------------------------------------------------------------------
+
+.. warning::
+   
+   We don't provide any scripts to automate this whole process, so please read
+   carefully and follow all the steps in order to obtain the desired results.
```
```suggestion
.. warning::
   
   We **do not** provide any scripts to automate this process, so please read
   carefully and follow all the steps to obtain the boot file.
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+
+When booting a system with Linux + HDL (depending on the project), you will need
+to write the following files on the root of the BOOT FAT32 partition. After that,
+you must write the preloader image in a specific card partition, depending on
```
```suggestion
you must write the **preloader image** in a specific card partition, depending on
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+   -  ``<target>/fit_spl_fpga.itb``
+   -  ``<target>/socfpga_arria10_socdk_sdmmc.dtb``
+   -  ``<target>/u-boot.img``
+   -  ``<target>/extlinux.conf → extlinux /extlinux.conf``
```
```suggestion
   -  ``<target>/extlinux.conf → extlinux/extlinux.conf``
```
I'm not familiar with this, but would this arrow look/work ok in the PDF documentation?

cristianmihaipopa replied:
Yes, it renders properly in PDF (tested). The characters that would usually mess up the PDF document are highlighted in Visual Studio Code with a yellow box. As far as I know by working with this, only in these cases the chars don't render properly. So the arrows will be kept.

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+   -  ``<target>/preloader_bootloader.img`` to the corresponding SD card
+      partition (e.g. ``dd if=preloader_bootloader.img of=/dev/mmcblk0p3``)
+
+-  For De10Nano SoC project:
```
Give link to the carriers like it was done here (in the code) https://analogdevicesinc.github.io/hdl/user_guide/architecture.html#intel-platforms

cristianmihaipopa replied:
Will update after we have a final version. Don't want to update now and need to update every time I add something new.

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+Examples of building the boot image
+-------------------------------------------------------------------------------
+
+Below is a list of projects supported by us for each carrier to illustrate how
+to build and showcase the different files that are involved in the process. Each
+project has it's own characteristics (some files may be different from a project
+to another).
```
```suggestion
This is a list of projects supported by us for each carrier. The purpose is to illustrate how
to build and showcase the different files that are involved in the process. Each
project has its own characteristics (some files may differ from one project
to the other).
```
Maybe the first line I edited can be split on the next one, to fit under the character limit per line (I can't see it here)

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+project has it's own characteristics (some files may be different from a project
+to another).
+
+.. note::
+
+   Pay attention, each project needs it's own specific Linux Kernel Image &
+   Devicetree to be build. Also, follow these instructions to write the file to
+   your SD card, depending on the operating system that you use (Windows or
+   Linux):
```
```suggestion
.. note::

   Each project has its own Linux Kernel Image &
   Devicetree to be build. Follow these instructions to write the file to
   your SD card, depending on the operating system that you use (Windows or
   Linux):
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,581 @@
+   user@analog:~$ git clone https://github.com/analogdevicesinc/linux.git
+   user@analog:~$ cd linux/
+   user@analog:~/linux$ export ARCH=arm
+   user@analog:~/linux$ export CROSS_COMPILE=/path/to/your/arm/cross-compiler
```
We should give a more specific path.
Maybe [this](https://www.intel.com/content/www/us/en/docs/programmable/683788/19-1/compiling-the-linux-kernel-for-the-intel.html) can help you (there it says CROSS_COMPILE=arm-linux-gnueabihf and I think the tool can find it, but it should be tested).

cristianmihaipopa replied:
Based on the provided page, I decided to rewrite it like this:
`CROSS_COMPILE=/path_to_cross_compiler/arm-linux-gnueabihf`

IuliaCMoldovan reviewed (changes requested):
A couple of comments, and more to follow as I didn't get to finish reviewing it.
Also, there are some " which I think will cause us trouble (build_intel_boot_bin.rst: 110-111)

cristianmihaipopa commented:
V2: Did the requested changes, left the FPGA name's tag unchanged until we have a final version of this page.

cristianmihaipopa commented:
V3: Patch applied and fixed some minor mistakes

gastmaier commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+:dokuwiki:`here <resources/tools-software/linux-software/adi-kuiper_images/release_notes>`.
+
+The files need to be copied to the BOOT FAT32 partition with the exception of
+the **preloader image**, which need needs to be written to the third partition
```
```suggestion
the **preloader image**, which needs to be written to the third partition
```

gastmaier commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+   $cp socfpga_arria10_socdk_ad9081/socfpga_arria10_socdk_sdmmc.dtb .
+   $cp socfpga_arria10_common/zImage .
+
+Writing the boot preloader partition require special attention,
```
```suggestion
Writing the boot preloader partition requires special attention,
```

gastmaier reviewed (changes requested):
Minor grammar fix + please rebase onto main

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+    ~/hdl/projects/adrv9371x/a10soc/software/bootloader
+   $cd u-boot-socfpga ; pwd
+    ~/hdl/projects/adrv9371x/a10soc/software/bootloader/u-boot-socfpga
+   $git checkout rel_socfpga_v2021.07_22.02.02_pr
```
For different Quartus versions, there needs to be checked out different u-boot branches.

cristianmihaipopa replied:
Put a general caution section were I explicitly highlighted the importance of checking out the proper u-boot branch based on Quartus version.

SRaus replied:
I'm ok with having "general caution section".

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+
+.. shell::
+
+   $time sudo dd if=./2023-12-13-ADI-Kuiper-full.img of=/dev/sdz status=progress bs=4194304
```
You can point here to Kuiper documentation (on wiki or analog.com for now) so people can use the latest.

cristianmihaipopa replied:
Already pointed in section 'Using Kuiper Linux pre-built images'. There's a link directly to Wiki Kuiper images page

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+   $    --settings software/bootloader/settings.bsp
+   $cd software/bootloader ; pwd
+    ~/hdl/projects/arradio/c5soc/software/bootloader
+   $git clone https://github.com/altera-opensource/u-boot-socfpga.git
```
Pay attention to the u-boot branch used - on de10nano example you check out a specific branch, but here not. 
For example 'socfpga_v2021.10' is used for Quartus 22.1 Std (so u-boot tag has nothing to do with Quartus version).

cristianmihaipopa replied:
This was taken from Wiki. I will fix it.

SRaus replied:
It should be corrected also on wiki, if the page will remain publicly accessible.

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+   $cd u-boot-socfpga ; pwd
+    ~/hdl/projects/arradio/c5soc/software/bootloader/u-boot-socfpga
+   $./arch/arm/mach-socfpga/qts-filter.sh cyclone5 ../../../../../board/altera/cyclone5-socdk/qts/
+   $make socfpga_arria10_defconfig
```
Here is Cyclone 5 case - > replace "socfpga_arria10_defconfig" with "socfpga_cyclone5_defconfig"

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+    ~/hdl/projects/arradio/c5soc/software/bootloader/u-boot-socfpga
+   $echo "load mmc 0:1 \${loadaddr} soc_system.rbf;" > u-boot.txt
+   $echo "fpga load 0 \${loadaddr} \$filesize;" >> u-boot.txt
+   $sed -i 's/ghrd_10as066n2/soc_system/g' board/altera/arria10-socdk/fit_spl_fpga.its
```
This line is specific to Arria10, there is no ITS file for Cyclone5.

cristianmihaipopa replied:
I'll remove the line.

cristianmihaipopa commented:
V4: Addressed the last issues mentioned in the PR. The typos were fixed. Regarding the rest of the comments, about the files & branches needed in the build process, I added several sections (like warning and cautions) to explain to the user that these tutorials used the available resources at that time and the user needs to make sure is responsible for the Quartus version and the branch for the u-boot image. The workflow remains the same & it's recommended to use the recent versions of Quartus.

SRaus commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,811 @@
+files:
+
+- For Arria 10 SoC projects (:ref:`build example <build_intel_boot_bin arria10>`):
+
```
ok

gastmaier reviewed (changes requested):
Minor changes with updated dd format partition command.

[0001-Review-Minor-fixes.patch](https://github.com/user-attachments/files/17491897/0001-Review-Minor-fixes.patch)

cristianmihaipopa commented:
V5: Final version. Rebased and merged the commits in one

ladace reviewed (dismissed):
No remarks found!

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,850 @@
+Building the Hardware Design
+```````````````````````````````````````````````````````````````````````````````
+
+Clone the HDL repository if it does not exist yet in local directory. Then build
````````````````````````````````````````````````````````````````````````````````
Same as lines 211-212

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+   $cd hdl/projects/arradio/c5soc
+   $make
+
+After the design was built, the resulted SRAM Object File (.sof) file shall be
+converted to a Raw Binary File (.rbf).
+
+If you skipped the last section, ensure to set the architecture and cross
+compiler environment variables.
+
+.. caution::
+
+   Pay attention to directory changes to where the commands are run from,
+   and always confirm with ``pwd`` to show the current path at you terminal.
```
Same as comments from lines 220+

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,850 @@
+Building the Preloader and Bootloader Image
+```````````````````````````````````````````````````````````````````````````````
+
+This flow applies starting with release 2021_R1 / Quartus Pro version 20.1.
+For older versions of the flow see previous versions of this page on wiki
+:dokuwiki:`Altera SOC Quick Start Guide <resources/tools-software/linux-software/altera_soc_images>`.
+
+In HDL project directory, create the software/bootloader folder and clone the
+``u-boot-socfpga`` image. Before that create a new BSP settings file:
````````````````````````````````````````````````````````````````````````````````
Same comments as for the previous example

IuliaCMoldovan replied:
Still present

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+   $echo "fpga load 0 \${loadaddr} \$filesize;" >> u-boot.txt
+   $./tools/mkimage -A arm -O linux -T script -C none -a 0 -e 0 -n "Cyclone V script" -d u-boot.txt u-boot.scr
+
+Last but not least, create the extlinux.conf linux configuration file. This
+extlinux folder shall be copied to /BOOT partition of the SD Card:
```
Same as for the previous example

IuliaCMoldovan replied:
Still present

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,850 @@
+Jumper setup
+```````````````````````````````````````````````````````````````````````````````
+
+Here is the jumper configuration to boot the image from the SD Card:
````````````````````````````````````````````````````````````````````````````````
\+ "for ARRADIO/C5SoC"

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+Here is the jumper configuration to boot the image from the SD Card:
+
+.. list-table::
+   :widths: 45 45
```
Should sum up to 100

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+   * - CODEC_SEL
+     - 0
+
+And set JP2 to 2.5V or 1.8V
```
```suggestion
And **set JP2 to 2.5V or 1.8V**.
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,850 @@
+Configuring the SD Card
+```````````````````````````````````````````````````````````````````````````````
+
+Below is the commands to create the preloader and bootloader partition using
````````````````````````````````````````````````````````````````````````````````
```suggestion
Below are the commands to create the preloader and bootloader partition using
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+
+.. _build_intel_boot_bin de10nano:
+
+CN0540/DE10Nano
```
Same comments as for the previous example

IuliaCMoldovan replied:
Still present. All the comments I had for the previous sections, they apply to this one as well, since it's the same text used for each example

IuliaCMoldovan reviewed (changes requested):
First things first, the name of the page should be changed. It is not the Intel BOOT.BIN, but the boot image as you mentioned in the title.

Also, forgot to leave review, but the last line from the file should be removed (it's an empty line).

cristianmihaipopa commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+Resolves inside the Makefiles as ``/path/to/arm-linux-gnueabihf-gcc``
+(added suffix ``gcc``).
+
+If your environment already have the compiler on path
+(simple test if :code:`which arm-linux-gnueabihf-gcc` returns the expected path),
+you can set ``CROSS_COMPILE`` simply to:
+
+.. shell::
+
+   $export CROSS_COMPILE=arm-linux-gnueabihf-
```
It's just another example where we assume that the user has a configured environment and doesn't need to put the whole path to the compiler.

IuliaCMoldovan replied:
```suggestion
If your environment already has the compiler in the path
(test if :code:`which arm-linux-gnueabihf-gcc` returns the expected path),
you can set ``CROSS_COMPILE`` to:

.. shell::

   $export CROSS_COMPILE=arm-linux-gnueabihf-
```

cristianmihaipopa commented on docs/user_guide/build_intel_boot_bin.rst:
```diff
@@ -0,0 +1,850 @@
+
+.. caution::
+
+   Pay attention to directoy changes to where the commands are run from,
+   and always confirm with ``pwd`` to show the current path at you terminal.
```
I wouldn't remove it, as it adds a degree of cautiones for the user. We just wat to emphasize on being very careful when following the build process. Small mistakes can interrupt or mess the whole process.

cristianmihaipopa commented:
V6: Did the requested changes that were mentioned after V5

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,852 @@
+Jumper setup
+```````````````````````````````````````````````````````````````````````````````
+
+Here is the jumper configuration for ARRADIO/C5SoCto to boot the image from the
````````````````````````````````````````````````````````````````````````````````
```suggestion
Here is the jumper configuration for ARRADIO/C5SoC to boot the image from the
```

IuliaCMoldovan replied:
```suggestion
Here is the jumper configuration for ARRADIO/C5SoC to boot the image from the
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,852 @@
+Configuring the SD Card
+```````````````````````````````````````````````````````````````````````````````
+
+Below is the commands to create the preloader and bootloader partition using
````````````````````````````````````````````````````````````````````````````````
Same

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,852 @@
+Building the Preloader and Bootloader Image
+```````````````````````````````````````````````````````````````````````````````
+
+This flow applies starting with release 2021_R1 / Quartus Pro version 20.1.
````````````````````````````````````````````````````````````````````````````````
Same

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
````````````````````````````````````````````````````````````````````````````````diff
@@ -0,0 +1,852 @@
+Building the Hardware Design
+```````````````````````````````````````````````````````````````````````````````
+
+Clone the HDL repository if it does not exist yet in local directory. Then build
````````````````````````````````````````````````````````````````````````````````
Same

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
```diff
@@ -0,0 +1,852 @@
+   $cd hdl/projects/cn0540/de10nano
+   $make
+
+After the design was built, the resulted SRAM Object File (.sof) file shall be
```
Same

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
```diff
@@ -0,0 +1,852 @@
+
+.. caution::
+
+   Pay attention to directoy changes to where the commands are run from,
```
```suggestion
   Pay attention to directory changes to where the commands are run from,
```

IuliaCMoldovan commented on docs/user_guide/build_intel_boot_image.rst:
```diff
@@ -0,0 +1,852 @@
+board).
+
+You may notice that in the ``export CROSS_COMPILE`` examples there is a
+"trailing" dash ``-``, and the reason for this is because an export:
+
+.. shell::
+
+   $export CROSS_COMPILE=/path/to/arm-linux-gnueabihf-
+
+Within the Makefiles, this path becomes /path/to/arm-linux-gnueabihf-gcc
+(with ``gcc`` added at the end).
```
```suggestion
"trailing" dash ``-``. That is because within the Makefiles, this path becomes
/path/to/arm-linux-gnueabihf-gcc (with ``gcc`` appended).

.. shell::

   $export CROSS_COMPILE=/path/to/arm-linux-gnueabihf-

```

cristianmihaipopa commented:
V6

IuliaCMoldovan reviewed (approved):
Just to squash the commits, and it all looks good
